### PR TITLE
Added the option to set overwrites on channel creation & Game+Status on StartAsync

### PR DESCRIPTION
**Pull request by davidsl4**
I have added the option to create permission overwrites when you create a new guild channel, suppose to [Discord Docs](https://discordapp.com/developers/docs/resources/guild#create-guild-channel) you can do it

### CODE REVIEW

File: src/Discord.Net.WebSocket/DiscordSocketClient.cs [hunk 1]
```diff
@@ -242,14 +261,27 @@ private async Task OnConnectingAsync()
...
+                    }).ConfigureAwait(false);
                 }
 
                 //Wait for READY
                 await _connection.WaitAsync().ConfigureAwait(false);
 
-                await _gatewayLogger.DebugAsync("Sending Status").ConfigureAwait(false);
-                await SendStatusAsync().ConfigureAwait(false);
+                /*await _gatewayLogger.DebugAsync("Sending Status").ConfigureAwait(false);
+                await SendStatusAsync().ConfigureAwait(false);*/
```
Comment (AraHaan) [inline src/Discord.Net.WebSocket/DiscordSocketClient.cs:284]: Why is this commented out?

Comment (AraHaan) [inline src/Discord.Net.WebSocket/DiscordSocketClient.cs:284]: Why is this commented out?

Comment (davidsl4) [reply src/Discord.Net.WebSocket/DiscordSocketClient.cs:284]:
```suggestion
if (Activity is RichGame)
                        throw new NotSupportedException("Outgoing Rich Presences are not supported via WebSocket.");
                    await ApiClient.SendIdentifyAsync(shardID: ShardId, totalShards: TotalShards, guildSubscriptions: _guildSubscriptions, presence: new StatusUpdateParams
                    {
                        Game = Activity == null ? null : new GameModel
                        {
                            Name = Activity.Name,
                            Type = Activity.Type,
                            StreamUrl = Activity is StreamingGame sg ? sg.Url : null,
                        },
                        Status = Status,
                        IsAFK = Status == UserStatus.AFK,
                        IdleSince = _statusSince != null ? _statusSince.Value.ToUnixTimeMilliseconds() : (long?)null
                    }).ConfigureAwait(false);
```
Same thing but on the way when sending the identify event, Discord supports setting status on identify event

Comment (AraHaan) [reply src/Discord.Net.WebSocket/DiscordSocketClient.cs:284]: @davidsl4 uh might want to rebase this pr to fix the merge conflicts.

File: src/Discord.Net.WebSocket/ConnectionManager.cs [hunk 2]
```diff
@@ -53,7 +54,7 @@ internal class ConnectionManager : IDisposable
             });
         }
 
-        public virtual async Task StartAsync()
+        public virtual async Task StartAsync(SocketPresence? presence = null)
```
Comment (xXBuilderBXx) [inline src/Discord.Net.WebSocket/ConnectionManager.cs:57]: This is not needed.

**Comment by SubZero0** [comment]
I'll be closing this one since there's no update to fix the merge conflicts or about the reviews.

**Pull request closed without merging**